analyses/compare-gistic/03-GISTIC-gene-level-tally.Rmd

```diff
@@ -0,0 +1,382 @@
+
+Note that we will prepare only files derived from the consensus SEG file.
+
+## Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/compare-gistic/03-GISTIC-gene-level-cn-status-comparison.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+```{r}
+library(tidyverse)
+```
+
+If this is running in CI, skip the LGAT steps because there will not be enough samples.
+
+```{r}
+if (params$is_ci == 0) {
+  run_lgat <- TRUE
+} else {
+  run_lgat <- FALSE
+}
+```
+
+### Functions
+
+For tallying up copy number status
+
+```{r}
+get_gene_tally <- function(gene_status_df,
+                           gene_mapping_df,
+                           method = "gistic",
+                           biospecimens_included = NULL,
+                           genes_included = NULL) {
+  # This function takes two data.frames: 1) either an annotated seg file 
+  # (method = "exon") from `focal-cn-file-preparation` or the gene status 
+  # data.frame from prepare_gene_level_gistic (method = "gistic") and 2) the 
+  # peak assignment output of prepare_gene_level_gistic and returns a tally of 
+  # copy number status for individual genes
+  # 
+  # Args:
+  #   gene_status_df: either an annotated seg file (set method = "exon") from 
+  #                   `focal-cn-file-preparation` or the gene status data.frame 
+  #                   from prepare_gene_level_gistic (set method = "gistic")
+  #   gene_mapping_df: the peak assignment output of prepare_gene_level_gistic 
+  #   method: "gistic" or "exon", see above
+  #   biospecimens_included: a vector of biospecimen identifiers to be used
+  #                          to filter gene_status_df; intended to be used when
+  #                          method = "exon"
+  #   genes_included: a vector of gene symbols to be used to filter 
+  #                   gene_status_df; intended to be used when method = "exon"
+  # 
+  # Returns:
+  #   gene_tally_df: a data.frame with the following columns
+  #                  1. gene: gene symbol
+  #                  2. gain: number of gains tallied
+  #                  3. loss: number of losses tallied
+  #                  4. neutral: number of copy neutral events tallied
+  #                  5. detection_peak: all peaks a gene symbol mapped to 
+  #                                     via GISTIC, comma-separated
+  
+  # TODO: error-handling throughout this function could probably be a bit more
+  # robust
+  if ((method == "exon") & any(c(is.null(biospecimens_included), 
+                                 is.null(genes_included)))) {
+    stop("The 'exon' method is intended to be used with the 
+         biospecimens_included and genes_included arguments")
+  }
+  
+  # If a vector of biospecimen IDs is provided, filter only to those biospecimen
+  # ids
+  if (!is.null(biospecimens_included)) {
```

Comment:
Is there an instance where one would use `biospecimens_included` but would *not* be using `method = "exon"`? The documentation makes me think no. If so, it might be better to separate these error handlers by `method`. It would make it easier to follow at least. However if there is an intention to allow biospecimen filtering for `method = "gistic"` then I would change the documentation to make that more clear and that it is just required for `method = "exon"`.

Reply:
Yeah, I saw that, but was wondering if that was something we could do now?